### PR TITLE
Clean up allocator test fixture names

### PR DESCRIPTION
**Pull request by burelc-amzn**
Now that allocators are always available, a test fixture to set them up is no longer necessary. The role of that test fixture is to check for memory leaks at the end of the test.

This renames the fixture to reflect its new role, and fixes some minor changes that were suggested in #12797 that weren't addressed before it was finished.

### CODE REVIEW

File: Code/Framework/AtomCore/Tests/ConcurrencyCheckerTests.cpp [hunk 1]
```diff
@@ -15,16 +15,16 @@ using namespace AZStd;
...
-        : public AllocatorsTestFixture
+        : public LeakDetectionFixture
     {
 
         void SetUp() override
         {
-            AllocatorsFixture::SetUp();
+            LeakDetectionFixture::SetUp();
         }
     };
```
Comment (amzn-phist) [inline Code/Framework/AtomCore/Tests/ConcurrencyCheckerTests.cpp:25]: Is this fixture not used?

File: Code/Framework/AzCore/Tests/AZStd/ConceptsTests.cpp [hunk 2]
```diff
@@ -67,11 +67,12 @@ namespace UnitTest
         // concept common_reference_with
         static_assert(AZStd::common_reference_with<TestBase&, Base&>);
         static_assert(AZStd::same_as<AZStd::common_reference_t<const TestBase&, Base&>, const Base&>);
-        static_assert(!AZStd::common_reference_with<AllocatorsTestFixture, ScopedAllocatorSetupFixture>);
+        // burelc FIXME
+        // static_assert(!AZStd::common_reference_with<LeakDetectionFixture, LeakDetectionFixture>);
```
Comment (amzn-phist) [inline Code/Framework/AzCore/Tests/AZStd/ConceptsTests.cpp:71]: Remove and file an issue.

File: Code/Framework/AzCore/Tests/AZStd/Pair.cpp [hunk 3]
```diff
@@ -17,17 +17,17 @@ namespace UnitTest
...
-            AllocatorsFixture::SetUp();
+            LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            AllocatorsFixture::TearDown();
+            LeakDetectionFixture::TearDown();
         }
     };
```
Comment (amzn-phist) [inline Code/Framework/AzCore/Tests/AZStd/Pair.cpp:32]: Instances of fixtures like this where the derived class doesn't do anything unique, can we simply omit these overrides of SetUp and TearDown?

File: Code/Framework/AzCore/Tests/RTTI/TypeSafeIntegralTests.cpp [hunk 4]
```diff
@@ -24,7 +24,7 @@ namespace UnitTest
 
     template <typename IntegralType>
     class TypeSafeIntegralTests
-        : public UnitTest::ScopedAllocatorSetupFixture
+        : public UnitTest::LeakDetectionFixture
     {
         ;
```
Comment (amzn-phist) [inline Code/Framework/AzCore/Tests/RTTI/TypeSafeIntegralTests.cpp:29]: Odd semicolon in the body of a class.  Remove it.

File: Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp [hunk 5]
```diff
@@ -338,8 +338,7 @@ namespace UnitTests
     }
 
     struct PathDependencyBenchmarks
-        : ::UnitTest::ScopedAllocatorFixture
-          , PathDependencyBase
+        : PathDependencyBase
     {
```
Comment (amzn-phist) [inline Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp:342]: This change removes the allocator fixture parent class, is it intentional?

File: Tools/LyTestTools/tests/integ/test_o3de_codeowners.py [hunk 6]
```diff
@@ -14,7 +14,7 @@
 def test_FindCodeowners_ThisCWD_FindForO3DE():
     codeowners_path = hint.find_github_codeowners(os.path.abspath(__file__))
     assert codeowners_path
-    assert "o3de" in str(codeowners_path), "codeowners file has been tampered with, or test unexpectedly migrated"
+    assert "main" in str(codeowners_path), "codeowners file has been tampered with, or test unexpectedly migrated"
```
Comment (amzn-phist) [inline Tools/LyTestTools/tests/integ/test_o3de_codeowners.py:17]: What's the motivation behind this change?  Seems like "o3de" would be found in a path rather than "main".

File: system_linux_pc.cfg [hunk 7]
```diff
@@ -22,3 +22,4 @@ r_ShadersAllowCompilation = 1
 
 -- Display FPS
 r_displayInfo = 3
+r_ShaderCompilerServer=127.0.0.1
```
Comment (amzn-phist) [inline system_linux_pc.cfg:25]: Is this change part of this PR?

File: Tools/LyTestTools/tests/integ/test_o3de_codeowners.py (hunk 6 above)
Comment (amzn-phist) [reply Tools/LyTestTools/tests/integ/test_o3de_codeowners.py:17]: I saw the linked GHI about this.  Would this change make other machines fail?  Seems like it's looking for a path fragment that's "above" or at the root of the repo, is that correct?  That could be different on any machine.

File: Code/Framework/AzCore/Tests/AZStd/Pair.cpp (hunk 3 above)
Comment (amzn-phist) [reply Code/Framework/AzCore/Tests/AZStd/Pair.cpp:32]: In general, there seems to be a lot of variance in how these fixture subclasses are written.  Some explicitly call SetUp and TearDown, like this one, others only call SetUp.  Some don't call SetUp/TearDown of the base class at all in their overrides.  I get that this is mostly a find-and-replace change, but if there are opportunities to remove unnecessary lines then I'm in favor.

File: system_linux_pc.cfg (hunk 7 above)
Comment (burelc-amzn) [reply system_linux_pc.cfg:25]: `git commit -a` strikes again. Will remove.

File: Code/Framework/AtomCore/Tests/ConcurrencyCheckerTests.cpp (hunk 1 above)
Comment (burelc-amzn) [reply Code/Framework/AtomCore/Tests/ConcurrencyCheckerTests.cpp:25]: Yes, I'll remove it.

File: Code/Framework/AzCore/Tests/RTTI/TypeSafeIntegralTests.cpp (hunk 4 above)
Comment (burelc-amzn) [reply Code/Framework/AzCore/Tests/RTTI/TypeSafeIntegralTests.cpp:29]: Fixed.

File: Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp (hunk 5 above)
Comment (burelc-amzn) [reply Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp]: It was conflicting with the base class of the `PathDependencyTestValidation` fixture. But this change was wrong anyway, I've added it back. and removed it from the more-derived type `PathDependencyTestValidation`.

File: Code/Framework/AzCore/Tests/AZStd/Pair.cpp (hunk 3 above)
Comment (burelc-amzn) [reply Code/Framework/AzCore/Tests/AZStd/Pair.cpp:32]: I used
```
grep -Pzo '(?s)void SetUp\(\) override\n *{ *\n *LeakDetectionFixture::SetUp\(\);\n *}\n' Code Gems -rIl
```

to find some 13 files that had empty `SetUp()` functions, and cleaned those up.

The `LeakDetectionFixture` does its work on construct/destruct, so there's no getting around those calls.

File: Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h [hunk 8]
```diff
@@ -2572,7 +2567,7 @@ namespace AZ
             return static_cast<GenericClassInfoType*>(findIt->second);
         }
 
-        void* rawMemory = m_moduleOSAllocator.Allocate(sizeof(GenericClassInfoType), alignof(GenericClassInfoType));
+        void* rawMemory = AllocatorInstance<OSAllocator>::Get().Allocate(sizeof(GenericClassInfoType), alignof(GenericClassInfoType));
```
[the remark this answers is not in the document]
Comment (burelc-amzn) [reply Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h:2570]: It can use either one. This is inserted into `m_moduleLocalGenericClassInfos`, which is a `AZStd::unordered_map<AZ::Uuid, AZ::GenericClassInfo*>`. That type's allocator ends up routing to the `SystemAllocator`, but that describes how the container itself allocates its internal data, not how the mapped type should be allocated. It's the same as having a `vector<SomeType>` where `SomeType` is defined to use the `PoolAllocator`. Since this was already being allocated with the `OSAllocator`, I kept that part.

File: Code/Framework/AzCore/Tests/AZStd/Pair.cpp (hunk 3 above)
Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/Tests/AZStd/Pair.cpp]: But the base class SetUp and TearDown functions will be called if the derived class implementations don't exist

File: Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h (hunk 8 above)
[the remark this answers is not in the document]
Comment (burelc-amzn) [reply Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h]: I've gone ahead and updated this to use `azcreate`, which will allocate the object using the `SystemAllocator`.

**Review by amzn-phist — APPROVED** [review]

File: Code/Framework/AzCore/Tests/AZStd/LockFreeQueues.cpp [hunk 9]
```diff
@@ -21,10 +21,10 @@ namespace UnitTest
     using namespace UnitTestInternal;
 
     class LockFreeQueue
-        : public AllocatorsFixture
+        : public LeakDetectionFixture
     {
     public:
-        LockFreeQueue() : AllocatorsFixture() {}
+        LockFreeQueue() : LeakDetectionFixture() {}
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/Tests/AZStd/LockFreeQueues.cpp:27]: You could just delete the constructor.

File: Code/Framework/AzCore/Tests/Slice.cpp [hunk 10]
```diff
@@ -200,17 +200,17 @@ namespace UnitTest
     };
 
     class SliceTest
-        : public AllocatorsFixture
+        : public LeakDetectionFixture
     {
     public:
         SliceTest()
-            : AllocatorsFixture()
+            : LeakDetectionFixture()
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/Tests/Slice.cpp:207]: This constructor can also be deleted.

File: Code/Framework/AzFramework/Tests/BinToTextEncode.cpp [hunk 11]
```diff
@@ -19,17 +19,17 @@ namespace UnitTest
 
     //! Unit Test for testing Base64 Encode/Decode functions
     class Base64Test
-        : public AllocatorsFixture
+        : public LeakDetectionFixture
     {
     public:
         Base64Test()
-            : AllocatorsFixture()
+            : LeakDetectionFixture()
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzFramework/Tests/BinToTextEncode.cpp:26]: This constructor can be deleted as well

File: Code/Framework/AzNetworking/Tests/Serialization/DeltaSerializerTests.cpp [hunk 12]
```diff
@@ -124,17 +124,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzNetworking/Tests/Serialization/DeltaSerializerTests.cpp:137]: Both the SetUp and TearDown functions can be deleted.

File: Code/Framework/AzToolsFramework/Tests/ActionManager/ActionManagerFixture.cpp [hunk 13]
```diff
@@ -49,7 +49,7 @@ namespace UnitTest
 
     void ActionManagerFixture::TearDown()
     {
-        AllocatorsTestFixture::TearDown();
+        LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/Tests/ActionManager/ActionManagerFixture.cpp:52]: Shouldn't this TearDown call be at the end of this function?

File: Code/Framework/AzToolsFramework/Tests/EntityTestbed.h [hunk 14]
```diff
@@ -67,7 +67,7 @@ namespace UnitTest
         AZ::IO::LocalFileIO m_localFileIO;
 
         EntityTestbed()
-            : AllocatorsFixture()
+            : LeakDetectionFixture()
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/Tests/EntityTestbed.h:70]: This constructor can be deleted.

File: Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp [hunk 15]
```diff
@@ -350,6 +349,8 @@ namespace UnitTests
         AzToolsFramework::AssetDatabase::ProductDatabaseEntry m_product1, m_product2, m_product4;
         AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntryContainer m_dependencies;
 
+        void TestBody() override {}
```
Comment (lumberyard-employee-dm) [inline Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp:352]: What is with the addition of the TestBody function?

File: Gems/Atom/Feature/Common/Code/Tests/SparseVectorTests.cpp [hunk 16]
```diff
@@ -16,17 +16,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/Feature/Common/Code/Tests/SparseVectorTests.cpp:29]: The Setup/TearDown function here can be deleted

File: Gems/Atom/Feature/Common/Code/Tests/MultiIndexedDataVectorTests.cpp [hunk 17]
```diff
@@ -19,17 +19,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/Feature/Common/Code/Tests/MultiIndexedDataVectorTests.cpp:32]: The SetUp/TearDown function here can be deleted

File: Gems/Atom/Feature/Common/Code/Tests/IndexedDataVectorTests.cpp [hunk 18]
```diff
@@ -18,17 +18,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/Feature/Common/Code/Tests/IndexedDataVectorTests.cpp:31]: The SetUp/TearDown function here can be deleted.

File: Gems/Atom/Feature/Common/Code/Tests/IndexableListTests.cpp [hunk 19]
```diff
@@ -21,17 +21,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/Feature/Common/Code/Tests/IndexableListTests.cpp:34]: The SetUp and TearDown function here can be deleted.

File: Gems/Atom/Feature/Common/Code/Tests/Decals/DecalTextureArrayTests.cpp [hunk 20]
```diff
@@ -24,17 +24,17 @@ namespace UnitTest
...
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         }
 
         void TearDown() override
         {
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/Feature/Common/Code/Tests/Decals/DecalTextureArrayTests.cpp:37]: The SetUp and TearDown function here can be deleted

File: Gems/Atom/RHI/Code/Tests/ContainerTests.cpp [hunk 21]
```diff
@@ -54,11 +54,11 @@ namespace UnitTest
     };
 
     class ChunkedVectorTest
-        : public UnitTest::AllocatorsFixture
+        : public UnitTest::LeakDetectionFixture
     {
     public:
         ChunkedVectorTest()
-            : AllocatorsFixture()
+            : LeakDetectionFixture()
```
Comment (lumberyard-employee-dm) [inline Gems/Atom/RHI/Code/Tests/ContainerTests.cpp:61]: The constructor can be deleted

File: Gems/EMotionFX/Code/Tests/CommandAdjustSimulatedObjectTests.cpp [hunk 22]
```diff
@@ -420,13 +420,13 @@ namespace EMotionFX
...
     class CommandAdjustSimulatedJointTestsFixture
-        : public UnitTest::AllocatorsTestFixture
+        : public UnitTest::LeakDetectionFixture
         , public ::testing::WithParamInterface<::testing::tuple<bool, bool, CommandAdjustSimulatedJointTestsParam>>
     {
     public:
         void SetUp() override
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
```
Comment (lumberyard-employee-dm) [inline Gems/EMotionFX/Code/Tests/CommandAdjustSimulatedObjectTests.cpp:429]: This SetUp function can be deleted.

File: Gems/EMotionFX/Code/Tests/SimulatedObjectSetupTests.cpp [hunk 23]
```diff
@@ -49,18 +49,18 @@ namespace SimulatedObjectSetupTests
...
         {
-            UnitTest::AllocatorsTestFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
             AZ::AllocatorInstance<ActorAllocator>::Create();
         }
         void TearDown() override
         {
             AZ::AllocatorInstance<ActorAllocator>::Destroy();
-            UnitTest::AllocatorsTestFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/EMotionFX/Code/Tests/SimulatedObjectSetupTests.cpp:63]: The SetUp and TearDown function here can be deleted

File: Gems/ScriptAutomation/Code/Tests/ScriptAutomationApplicationFixture.cpp [hunk 24]
```diff
@@ -20,7 +20,7 @@ namespace UnitTest
 {
     void ScriptAutomationApplicationFixture::SetUp()
     {
-        AllocatorsFixture::SetUp();
+        LeakDetectionFixture::SetUp();
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptAutomation/Code/Tests/ScriptAutomationApplicationFixture.cpp:23]: The SetUp function can be deleted

File: Gems/ScriptCanvas/Code/Tests/Framework/ScriptCanvasUnitTestFixture.h [hunk 25]
```diff
@@ -12,17 +12,17 @@
...
         void SetUp() override
         {
-            UnitTest::AllocatorsFixture::SetUp();
+            UnitTest::LeakDetectionFixture::SetUp();
         };
 
         void TearDown() override
         {
-            UnitTest::AllocatorsFixture::TearDown();
+            UnitTest::LeakDetectionFixture::TearDown();
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptCanvas/Code/Tests/Framework/ScriptCanvasUnitTestFixture.h:25]: The SetUp and TearDown function here can be deleted

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by Kadino — APPROVED** [review]
Rename and cleanup looks good. @lumberyard-employee-dm raises an interesting point above, which could eventually allow a crash during test execution.

No concerns from @o3de/sig-testing

File: Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp (hunk 15 above)
Comment (burelc-amzn) [reply Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp]: I kept getting confused by the inheritance structure of the previous class layout. I originally changed `PathDependencyBenchmarks` to now inherit from `LeakDetectionFixture`, which inherits from `testing::Test`, which was not in the inheritance chain before. But then when it is `new`'d:

https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp#L465

it is missing an implementation for `TestBody()`, so that doesn't work. When I changed `PathDependencyBenchmarks` to inherit from `LeakDetectionBase` instead, then the `PathDependencyTestValidation` class was malformed, because I had removed the `testing::Test` base and forgot to add it back:

https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/tests/PathDependencyManagerTests.cpp#L492-L495

My initial reaction was to just add an empty `TestBody()` method, but that's all fixed in the latest commit.

**Review by lumberyard-employee-dm — APPROVED** [review]
I am good with the changes now
